packages/funding-rate-proposer/src/proposer.js

```diff
@@ -153,57 +157,34 @@ class FundingRateProposer {
 
     // Assume pricefeed has been cached and updated prior to this function via the `update()` call.
     const priceFeed = this.priceFeedCache[fundingRateIdentifier];
-    if (!priceFeed) {
-      this.logger.error({
-        at: "PerpetualProposer#updateFundingRate",
-        message: "Failed to create pricefeed for funding rate identifier",
-        fundingRateIdentifier
-      });
-      return;
-    }
-    let offchainFundingRate = priceFeed.getCurrentPrice();
-    if (!offchainFundingRate) {
-      this.logger.error({
-        at: "PerpetualProposer#updateFundingRate",
-        message: "Failed to query current price for funding rate identifier",
-        fundingRateIdentifier
-      });
-      return;
-    }
-    // Set `offchainFundingRate` to correct precision by converting back and forth between the pricefeed's output
-    // precision:
-    offchainFundingRate = parseFixed(
-      Number(formatFixed(offchainFundingRate.toString(), priceFeed.getPriceFeedDecimals()))
-        .toFixed(this.precision)
-        .toString(),
-      priceFeed.getPriceFeedDecimals()
-    ).toString();
+    const pricefeedPrice = await this._getPriceFeedAndPrice(fundingRateIdentifier);
+    if (!pricefeedPrice) return;
```

Comment:
Why did we drop the error here?

Reply:
I refactored the error into an internal helper function